l10n_jp_summary_invoice/tests/test_l10n_jp_summary_invoice.py

```diff
@@ -0,0 +1,124 @@
+                        {
+                            "product_id": self.product.id,
+                            "quantity": 1,
+                            "price_unit": amount,
+                            "tax_ids": [Command.set(tax.ids)],
+                        }
+                    )
+                ],
+            }
+        )
+        invoice.action_post()
+        return invoice
+
+    def test_constrains_invoice_not_for_billing(self):
+        invoice = self._create_invoice(100, self.tax_15)
+        invoice.write({"is_not_for_billing": True})
+
+        with self.assertRaises(ValidationError):
+            self.env["account.billing"].create(
+                {
+                    "partner_id": self.partner.id,
+                    "billing_line_ids": [Command.create({"move_id": invoice.id})],
+                }
+            )
+
+    def test_constrains_remit_to_bank_conflict(self):
+        invoice = self._create_invoice(100, self.tax_15, bank=self.bank_account)
+        other_bank = self.env["res.partner.bank"].create(
+            {
+                "partner_id": self.env.company.partner_id.id,
+                "acc_number": "other_bank_acc",
+            }
+        )
+
+        with self.assertRaises(ValidationError):
+            self.env["account.billing"].create(
+                {
+                    "partner_id": self.partner.id,
+                    "remit_to_bank_id": other_bank.id,
+                    "billing_line_ids": [Command.create({"move_id": invoice.id})],
+                }
+            )
+
+    def test_compute_billing_due_date(self):
+        inv1 = self._create_invoice(100, self.tax_15)
+        inv2 = self._create_invoice(200, self.tax_15)
+        inv1.invoice_date_due = inv1.invoice_date_due.replace(day=5)
+        inv2.invoice_date_due = inv2.invoice_date_due.replace(day=25)
+        billing = self.env["account.billing"].create(
+            {
+                "partner_id": self.partner.id,
+                "billing_line_ids": [
+                    Command.create({"move_id": inv1.id}),
+                    Command.create({"move_id": inv2.id}),
+                ],
+            }
+        )
+        self.assertEqual(billing.date_due, inv2.invoice_date_due)
+
+    def test_update_remit_to_bank_defaulting(self):
+        invoice = self._create_invoice(100, self.tax_15, bank=self.bank_account)
+        billing = self.env["account.billing"].create(
+            {
+                "partner_id": self.partner.id,
+                "billing_line_ids": [Command.create({"move_id": invoice.id})],
+            }
+        )
+        self.assertEqual(billing.remit_to_bank_id, self.bank_account)
+
+    def test_get_moves_filters_billed_and_flags(self):
+        billed_invoice = self._create_invoice(50, self.tax_15)
+        billed_invoice.write({"is_not_for_billing": True})
+        billing = self.env["account.billing"].create(
+            {
+                "partner_id": self.partner.id,
+            }
+        )
```

Comment:
To save a few lines.
```suggestion
        billing = self.env["account.billing"].create({"partner_id": self.partner.id})
```